junit-engine-api/src/main/java/org/junit/gen5/engine/TestDescriptor.java

```diff
@@ -44,11 +40,16 @@ default boolean isRoot() {
 
 	Set<TestTag> getTags();
 
-	void addChild(TestDescriptor descriptor);
+	Set<? extends TestDescriptor> getChildren();
 
-	void removeChild(TestDescriptor descriptor);
-
-	Set<TestDescriptor> getChildren();
+	default Set<TestDescriptor> allDescendants() {
```

Comment:
I've used `Set<? extends TestDescriptor>` for methods that are overridden in `MutableTestDescriptor`. I wasn't sure where `allDescendants` will be used. Probably only through `MutableTestDescriptor`? In this case, it could even return `Set<MutableTestDescriptor>`.

Anyway, I'll change it for consistency now... whereever it may end up later on.